fairlearn/metrics/_metric_frame.py

```diff
@@ -783,20 +845,25 @@ def _process_features(self, base_name, features, sample_array) -> List[GroupFeat
 
         return result
 
-    def _mask_from_tuple(self, index_tuple, feature_list) -> np.ndarray:
-        """Generate a mask for the ``y_true``, ``y_pred`` and ``sample_params`` arrays.
+    def _accrete_array(
```

Comment:
Mhmm... this seems to be a complicated mechanism for this. I had in mind more something along the following lines (i.e., splitting into rows rather than columns):

```python
y_true=np.array([[0, 1], [1, 0], [0, 1]])
y_pred=np.array([[0, 1], [0, 1], [1, 0]])
sf = ['a','a','b']

df = pd.DataFrame({"sf": sf, "y_true": list(y_true), "y_pred": list(y_pred)})
df_accuracy_score = lambda df: skm.accuracy_score(list(df['y_true']), list(df['y_pred']))
by_group = df.groupby('sf').apply(df_metric)
```

Reply:
I've incorporated this, since it slightly simplifies the handling of multi-dimensional columns. I just hope that the extra type change doesn't have some other subtle side effect :-/